test/test_string_utils.cpp

```diff
@@ -0,0 +1,47 @@
+/* Copyright (C) 2021, Nikolai Wuttke. All rights reserved.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <base/warnings.hpp>
+#include <common/string_utils.hpp>
+
+RIGEL_DISABLE_WARNINGS
+#include <catch.hpp>
+RIGEL_RESTORE_WARNINGS
+
+TEST_CASE("String split") {
+  SECTION("Empty string") {
+    std::vector<std::string> v;
+    rigel::strings::split(v, "", ' ');
+    CHECK(v.size() == 1);
+  }
+  SECTION("String with no delimiter") {
+    std::vector<std::string> v;
+    rigel::strings::split(v, "hello", ' ');
+    CHECK(v.size() == 1);
+  }
+  SECTION("String with delimiter") {
+    std::vector<std::string> v;
+    rigel::strings::split(v, "hello, world", ',');
+    CHECK(v.size() == 2);
```

Comment:
Suggestion for making the test assertions a bit nicer:

```cpp
const auto expected = std::vector<std::string>{"hello", " world"};
CHECK(v == expectedResult);
```

That way, both the length of the result as well as the values are checked in one statement.

You could also use deduction guides and literal postfix to make it even shorter:

```cpp
using namespace std::string_literals;
const auto expected = std::vector{"hello"s, " world"s};
```

Up to you which one you prefer, both work for me 🙂

Reply:
Sure thing, Have to admit I've not used Catch in a while!
CTAD is a thing of beauty, but somehow I prefer the first one (call me old fashion), and if both are OK with you I'll go with it.